packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -288,157 +296,124 @@ export class OdspDocumentService implements IDocumentService {
      * @param token - authorization token for storage service
      * @param io - websocket library
      * @param client - information about the client
-     * @param url - websocket URL
-     * @param url2 - alternate websocket URL
+     * @param nonAfdUrl - websocket URL
+     * @param afdUrl - alternate websocket URL
      */
     private async connectToDeltaStreamWithRetry(
         tenantId: string,
         documentId: string,
         token: string | null,
         io: SocketIOClientStatic,
         client: IClient,
-        url: string,
-        url2?: string): Promise<IDocumentDeltaConnection> {
-        const hasUrl2 = !!url2;
-
-        // Create null logger if telemetry logger is not available from caller
-        const logger = this.logger ? this.logger : new TelemetryNullLogger();
-
-        let afdCacheValid = false;
-
-        if (this.localStorageAvailable) {
-            const lastAfdConnection = localStorage.getItem(lastAfdConnectionTimeMsKey);
-            if (lastAfdConnection !== null) {
-                const lastAfdTimeMs = Number(lastAfdConnection);
-                // If we have used the AFD URL within a certain amount of time in the past,
-                // then we should use it again.
-                if (!isNaN(lastAfdTimeMs) && lastAfdTimeMs > 0
-                    && Date.now() - lastAfdTimeMs <= afdUrlConnectExpirationMs) {
-                    afdCacheValid = true;
-                } else {
-                    localStorage.removeItem(lastAfdConnectionTimeMsKey);
-                }
-            }
-        }
-
-        // Use AFD URL if in cache
-        if (afdCacheValid && hasUrl2) {
-            debug("Connecting to AFD URL directly due to valid cache.");
-            const startAfd = performance.now();
-
-            return OdspDocumentDeltaConnection.create(
-                tenantId,
-                documentId,
-                token,
-                io,
-                client,
-                url2!,
-                20000,
-                this.logger,
-            ).then((connection) => {
-                logger.sendTelemetryEvent({
-                    eventName: "UsedAfdUrl",
-                    fromCache: true,
+        nonAfdUrl: string,
+        afdUrl?: string,
+    ): Promise<IDocumentDeltaConnection> {
+        const connectWithNonAfd = async () => {
+            const startTime = performance.now();
+            try {
+                const connection = await OdspDocumentDeltaConnection.create(
+                    tenantId,
+                    documentId,
+                    token,
+                    io,
+                    client,
+                    nonAfdUrl,
+                    20000,
+                    this.logger,
+                );
+                const endTime = performance.now();
+                this.logger.sendPerformanceEvent({
```

Comment:
Nit: should we think about some wrapper like PerformanceEvent.timedExecAsync() for simple cases like this? I.e. same as timedExecAsync but without start event? Maybe an optional argument for timedExecAsync.
catch() blocks makes it a bit tricky and code would be rather similar to what it is today, but it feels like it decreases chances of developers to screwed up telemetry.

Reply:
I'm not really a fan of `timedExecAsync` since it shows up on the stack.  One thing I did experiment with was using `performance.mark()`, `performance.measure()`, and `PerformanceObserver` to standardize telemetry measuring plus also add performance trace data in dev tools.  Maybe something like adding `startMeasure()` + `endMeasure()` helpers.  I decided that probably deserves more in-depth thought than I want to put into this PR though.